core/src/query/cursor.rs

```diff
@@ -128,4 +133,11 @@ impl QueryBatchedErasedIterator {
     ) -> Result<(QueryOutputBatchBox, Option<NonZeroU64>), UnknownCursor> {
         self.inner.next_batch(cursor)
     }
+
+    /// Returns the number of remaining elements in the iterator.
+    ///
+    /// NOTE: this is not a safety guarantee, same as it relies on [`ExactSizeIterator::len`].
```

Comment:
what does this mean?

Reply:
It means that you cannot rely on this reporting correct value for safety, it is safe to have an `ExactSizeIterator` reporting incorrect values. Maybe I should formulate this differently...

Reply:
Should be more clear now